lib/puppet/type/iis_virtual_directory.rb

```diff
@@ -46,6 +47,14 @@
     end
   end
 
+  newproperty(:user_name, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+    desc "Specifies the identity that should be impersonated when accessing the physical path."
+  end
+
+  newproperty(:password, :parent => PuppetX::PuppetLabs::IIS::Property::String) do
+    desc "Specifies the password associated with the user_name property."
```

Comment:
Should we add a step to the validate block to ensure a `password` is supplied if `user_name` is specified?

Reply:
We could ... like the following?

```
  validate do
    if (self[:user_name].nil? or self[:user_name].empty?) && self[:password]
      raise ArgumentError, "A user_name is required when specifying password."
    end
    if (self[:password].nil? or self[:password].empty?) && self[:user_name]
      raise ArgumentError, "A password is required when specifying user_name."
    end
  end
```

Reply:
Something more like line 103

Reply:
But this file has only 61 lines? I'm following this to perform validation across multiple properties:  https://github.com/puppetlabs/puppet-specifications/blob/master/language/resource_types.md#validate

Reply:
I can't count and doing too many PR reviews at once. I'm fine with your suggestion. If you can't add it today, I'll add it so we can close out the 4.2.1 release this week

Reply:
Word. 
Added.